api/dao/documentDAO.js

```diff
@@ -0,0 +1,454 @@
+const defaultLog      = require('winston').loggers.get('default');
+const mongoose        = require('mongoose');
+const Actions         = require('../helpers/actions');
+const Utils           = require('../helpers/utils');
+const intformat       = require('biguint-format');
+const FlakeIdGen      = require('flake-idgen');
+const uploadDir       = process.env.UPLOAD_DIRECTORY || "./uploads/";
+const MinioController = require('../helpers/minio');
+const constants       = require('../helpers/constants');
+const mime            = require('mime-types');
+const fs              = require('fs');
+
+const generator = new FlakeIdGen;
+const ENABLE_VIRUS_SCANNING = process.env.ENABLE_VIRUS_SCANNING || false;
+
+exports.documentHateoas = function(document, roles)
+{
+    document.links = 
+    [
+        { rel: 'self', title: 'public self', type: 'GET', href: '/api/v2/Public/Documents/' + document._id },
+        { rel: 'fetch', title: 'Public Document Download', method: 'GET', href: '/api/v2/Public/Documents/' + document._id + '/Download' }
+    ];
+
+    if (roles && roles.length > 0 && (roles.includes('sysadmin') || roles.includes('staff')))
+    {
+        document.links.push({ rel: 'self', title: 'secure self', method: 'GET', href: '/api/v2/Documents/' + document._id });
+        document.links.push({ rel: 'update', title: 'secure Document Update', method: 'PUT', href: '/api/v2/Documents/' + document._id });
+        document.links.push({ rel: 'delete', title: 'secure Document Delete', method: 'DELETE', href: '/api/v2/Documents/' + document._id });
+        document.links.push({ rel: 'update', title: 'secure Document Publish', method: 'GET', href: '/api/v2/Documents/' + document._id + '/Publish' });
+        document.links.push({ rel: 'update', title: 'secure Document UnPublish', method: 'GET', href: '/api/v2/Documents/' + document._id + '/Unpublish' });
+        document.links.push({ rel: 'update', title: 'secure Document Feature', method: 'GET', href: '/api/v2/Documents/' + document._id + '/Feature'});
+        document.links.push({ rel: 'update', title: 'secure Document UnFeature', method: 'GET', href: '/api/v2/Documents/' + document._id + '/Unfeature' });
+        document.links.push({ rel: 'fetch', title: 'secure Document Download', method: 'GET', href: '/api/v2/Documents/' + document._id + '/Download' });
+    }
+
+    return document;
+};
+
+exports.createDocument = async function(userName, projectId, comment, uploadedFile, ext, documentDetails, isPublic)
+{
+    let guid = intformat(generator.next(), 'dec');
+    let tempFilePath = uploadDir + guid + "." + ext;
+
+    try 
+    {
+        let virusScanSuccessful = false;
```

Comment:
Could just set to `true` and remove the `else` statement.

Reply:
Done!